File: c++/FilmTayang.cpp
#include "Film.cpp"

class FilmTayang : public Film {

private:
    // Nama bioskop
    string namaBioskop;
    // Lokasi bioskop
    string lokasi;
    // Rating usia
    string ratingUsia;

public:
        // Constructor kosong
    FilmTayang(): Film(){
        namaBioskop = "";
        lokasi = "";
        ratingUsia = "";
    }
    
    // Constructor FilmTayang
    FilmTayang(string id, string judul, string genre, int durasi, string namaBioskop, string lokasi, string ratingUsia):

    // Memanggil constructor Film
    Film(id, judul, genre, durasi){

        // Mengisi nama bioskop
        this->namaBioskop = namaBioskop;
        // Mengisi lokasi
        this->lokasi = lokasi;
        // Mengisi rating
        this->ratingUsia = ratingUsia;
    }

    // Getter nama bioskop
    string getNamaBioskop(){
        return namaBioskop;
    }
    // Setter nama bioskop
    void setNamaBioskop(string namaBioskop){
        this->namaBioskop = namaBioskop;
    }

    // Getter lokasi
    string getLokasi(){
        return lokasi;
    }
    // Setter lokasi
    void setLokasi(string lokasi){
        this->lokasi = lokasi;
    }

    // Getter rating
    string getRatingUsia(){
        return ratingUsia;
    }
    // Setter rating
    void setRatingUsia(string ratingUsia){
        this->ratingUsia = ratingUsia;
    }
};

File: php/FilmTayang.php
<?php

// Memanggil parent class
require_once "Film.php";

// Membuat class FilmTayang
class FilmTayang extends Film{
    // Atribut tambahan
    private $namaBioskop;
    private $lokasi;
    private $ratingUsia;

    // Constructor FilmTayang
    public function __construct($poster_film, $id, $judul, $genre, $durasi, $namaBioskop, $lokasi, $ratingUsia){
        // Memanggil constructor parent Film
        parent::__construct($poster_film, $id, $judul, $genre, $durasi);

        // Mengisi atribut tambahan
        $this->namaBioskop = $namaBioskop;
        $this->lokasi = $lokasi;
        $this->ratingUsia = $ratingUsia;
    }

    // Getter nama bioskop
    public function getNamaBioskop(){
        return $this->namaBioskop;
    }
    // Setter nama bioskop
    public function setNamaBioskop($namaBioskop){
        $this->namaBioskop = $namaBioskop;
    }

    // Getter lokasi
    public function getLokasi(){
        return $this->lokasi;
    }
    // Setter lokasi
    public function setLokasi($lokasi){
        $this->lokasi = $lokasi;
    }

    // Getter rating usia
    public function getRatingUsia(){
        return $this->ratingUsia;
    }
    // Setter rating usia
    public function setRatingUsia($ratingUsia){
        $this->ratingUsia = $ratingUsia;
    }
}
?>
